File: day 09/day9.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace AdventProject
{
  class AdventOfCode
  {
    static void Main(string[] args)
    {
      string inputFile = Directory.GetParent(Directory.GetCurrentDirectory()).Parent.Parent.FullName + @"\in.txt";
      StreamReader cin = new StreamReader(inputFile);

      int preamble = 25;
      List<long> numbers = new List<long>();
      long n, solution = 0;
      int offset = 0;

      for (int i = 0; i < preamble; i++)
      {
        n = long.Parse(cin.ReadLine());
        numbers.Add(n);
      }

      offset = 0;
      while (!cin.EndOfStream)
      {
        string x = (cin.ReadLine());

        n = long.Parse(x);

        bool valid = false;
        for (int i = offset; i < numbers.Count; i++)
        {
          for (int j = i + 1; j < numbers.Count; j++)
          {
            if (numbers[i] + numbers[j] == n)
            {
              valid = true;
            }
          }
        }

        offset++;
        numbers.Add(n);

        if (!valid)
        {
          Console.WriteLine("Part 1: " + n);
          solution = n;
        }
      }

      for (int i = 0; i < numbers.Count; i++)
      {
        long sum = 0, Min = 9999999, Max = 0;
        for (int j = i; j < numbers.Count && sum < solution; j++)
        {
          Min = Math.Min(numbers[j], Min);
          Max = Math.Max(numbers[j], Max);
          sum += numbers[j];
          if (sum == solution && numbers[j] != solution)
          {
            Console.WriteLine("Part 2: " + (Min + Max));
          }
        }
      }
    }
  }
}

File: day 09/day9.java
import java.io.File;
import java.io.IOException;
import java.util.*;

public class AdventOfCode {

	public static void main(String[] args) throws IOException {

		File file = new File("D:\\Java\\AdventOfCode\\in.txt");
		Scanner cin = new Scanner(file);

		int preamble = 25;
		Vector<Integer> numbers = new Vector<Integer>();
		int n, offset = 0, solution = 0;

		for (int i = 0; i < preamble; i++) {
			n = cin.nextInt();
			numbers.add(n);
		}

		offset = 0;
		while (cin.hasNextInt()) {
			{
				n = cin.nextInt();

				boolean valid = false;
				for (int i = offset; i < numbers.size(); i++) {
					for (int j = i + 1; j < numbers.size(); j++) {
						if (numbers.get(i) + numbers.get(j) == n) {
							valid = true;
						}
					}
				}

				offset++;
				numbers.add(n);

				if (!valid) {
					System.out.println("Part 1: " + n);
					solution = n;
				}
			}

			for (int i = 0; i < numbers.size(); i++) {
				int sum = 0, Min = 1999999999, Max = 0;
				for (int j = i; j < numbers.size() && sum < solution; j++) {
					Min = Math.min(numbers.get(j), Min);
					Max = Math.max(numbers.get(j), Max);
					sum += numbers.get(j);
					if (sum == solution && numbers.get(j) != solution) {
						System.out.println("Part 2: " + (Min + Max));
						System.exit(0);
					}
				}
			}
		}

	}
}
